src/components/file-handler/FileHandler.jsx

```diff
@@ -0,0 +1,167 @@
+import classNames from 'classnames';
+
+import Text from '../text/Text';
+import Link from '../text/Link';
+import Icon from '../icons/Icon';
+import Spinner from '../spinner/Spinner';
+
+import type {IconTypeType} from '../icons/Icon';
+
+export type FileHandlerColorType = 'gray' | 'mono';
+
+export const COLORS_MAP = {
+  gray: 'gray-secondary-light',
+  mono: 'white',
+};
+
+export const FILE_HANDLER_COLORS_SET = {
+  GRAY: 'gray',
+  MONO: 'mono',
+};
+
+type PropsType = $ReadOnly<{
+  /**
+   * Specify color of the background for FileHandler
+   * @example <FileHandler color="mono">
+   *            text
+   *          </FileHandler>
+   * @default gray
+   */
+  color: FileHandlerColorType,
+  /**
+   * Specify iconType to display SG icon as the image inside FileHandler
+   * @example <FileHandler iconType="answer">
+   *            text
+   *          </FileHandler>
+   * @default attachment
+   */
+  iconType?: IconTypeType,
+  /**
+   * Specify thumbnailSrc of the image inside FileHandler
+   * @example <FileHandler thumbnailSrc="https://source.unsplash.com/64x64/?bird">
+   *            text
+   *          </FileHandler>
+   */
+  thumbnailSrc?: string,
+  /**
+   * Specify src of the file to display it in the new tab when link is clicked
+   * @example <FileHandler src="https://source.unsplash.com/64x64/?bird">
+   *            text
+   *          </FileHandler>
+   */
+  src?: string,
+  /**
+   * Optional boolean for loading state of FileHandler
+   * @example <FileHandler loading>
+   *            text
+   *          </FileHandler>
+   * @default false
+   */
+  loading?: boolean,
+  /**
+   * Callback, called by clicking on **close** button. If specified, button will be added automatically
+   * @example <FileHandler onClose={() => doSomething()}>
+   *            text
+   *          </FileHandler>
+   */
+  onClose?: (SyntheticMouseEvent<HTMLDivElement>) => mixed,
+  /**
+   * Callback, called by clicking on link
+   * @example <FileHandler
+   *           src="https://source.unsplash.com/64x64/?bird"
+   *           onClose={() => doSomething()}
+   *          >
+   *            text
+   *          </FileHandler>
+   */
+  onClick?: () => mixed,
+  /**
+   * Additional function to set ref for text
+   */
+  setTextRef?: (ref: ?HTMLElement) => mixed,
+  /**
+   * Additional class names
+   */
+  className?: string,
+  /**
+   * Children to be rendered inside FileHandler
+   * @example <FileHandler>
+   *            text
+   *          </FileHandler>
+   */
+  children: Node,
+  ...
+}>;
+
+const FileHandler = ({
+  children,
+  color = 'gray',
+  iconType = 'attachment',
+  thumbnailSrc,
+  src,
+  loading = false,
+  onClose,
+  onClick,
+  setTextRef,
+  className,
+  ...props
+}: PropsType) => {
+  const fileHandlerClass = classNames('sg-file-handler', {
+    'sg-file-handler--closable': onClose,
+    [`sg-file-handler--${COLORS_MAP[color]}`]: color,
+  });
+
+  const showLink = src !== null && src !== undefined;
```

Comment:
Not sure if null check if necessary: `Optional parameters will accept missing, undefined, or matching types. But they will not accept null.` from https://flow.org/en/docs/types/functions/

Reply:
you are right, thank you! ❤️